File: spring-docs-viewer/src/main/java/com/docs/viewer/global/error/dto/response/ErrorHandler.java
package com.docs.viewer.global.error.dto.response;

import com.docs.viewer.global.error.dto.ErrorCode;
import com.docs.viewer.global.error.dto.ErrorInfo;
import org.springframework.stereotype.Component;

@Component
public class ErrorHandler {

    public ErrorResponse buildError(ErrorCode errorCode, ErrorInfo errorMessage) {
        return ErrorResponse.builder()
                .code(errorCode.getCode())
                .status(errorCode.getStatus())
                .message(errorCode.getMessage())
                .errors(errorMessage)
                .build();
    }
}

File: dotnet-docs-viewer/Common/Error/Response/ErrorHandler.cs
using WebApp.Common.Error.Types;

namespace WebApp.Common.Error
{
    public interface IErrorHandler
    {
        ErrorResponse BuildError(ErrorCode errorCode, ErrorInfo errorMessage);
    }
    
    public class ErrorHandler : IErrorHandler
    {
        public ErrorResponse BuildError(ErrorCode errorCode, ErrorInfo errorMessage)
        {
            var meta = ErrorCodeMetadata.Get(errorCode);

            return new ErrorResponse
            {
                Code = meta.Code,
                Status = meta.Status,
                Message = meta.Message,
                Errors = errorMessage
            };
        }
    }
}